src/models/invites-ignorer.ts

```diff
@@ -227,21 +255,48 @@ export class IgnoredInvites {
             const room = this.client.getRoom(target);
             if (room) {
                 return room;
-            } else {
-                target = null;
             }
         }
-        // We need to create our own policy room for ignoring invites.
-        target = (await this.client.createRoom({
-            name: "Individual Policy Room",
-            preset: Preset.PrivateChat,
-        })).room_id;
-        await this.withIgnoreInvitesPolicies(ignoreInvitesPolicies => {
-            ignoreInvitesPolicies.target = target;
-        });
 
-        // Since we have just called `createRoom`, `getRoom` should not be `null`.
-        return this.client.getRoom(target)!;
+        // If we reach this line, we need to setup the target room.
+        // However, it is possible that other callers within this client
+        // may be racing with us.
+        if (this.getOrCreateTargetRoomPromise) {
+            // Another caller is already calling this method.
+            // Merge the calls.
+            return this.getOrCreateTargetRoomPromise;
+        }
+        try {
+            // Nobody is calling the method at the moment.
+            // Register ourselves as the leader and start creating our policy room.
+            this.getOrCreateTargetRoomPromise = (async () => {
+                // We need to create our own policy room for ignoring invites.
+                target = (await this.client.createRoom({
+                    name: "Individual Policy Room",
```

Comment:
Should this be translated?

Reply:
Oh, good point. Is there any way to do that in matrix-js-sdk? Or should I somehow pass that string from react-sdk?